shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/API/Mempool.hs

```diff
@@ -33,13 +31,57 @@ import Control.State.Transition.Extended
     applySTS,
   )
 import Data.Sequence (Seq)
+import Data.Typeable (Typeable)
+import NoThunks.Class (NoThunks)
 import Shelley.Spec.Ledger.API.Validation
 import Shelley.Spec.Ledger.BaseTypes (Globals)
+import Shelley.Spec.Ledger.Keys (DSignable)
 import qualified Shelley.Spec.Ledger.LedgerState as LedgerState
 import Shelley.Spec.Ledger.STS.Ledgers (LEDGERS)
 import qualified Shelley.Spec.Ledger.STS.Ledgers as Ledgers
 import Shelley.Spec.Ledger.Slot (SlotNo)
 import Shelley.Spec.Ledger.Tx (Tx)
+import Shelley.Spec.Ledger.TxBody (EraIndependentTxBody)
+
+-- TODO #1304: add reapplyTxs
+class
+  ( Eq (Tx era),
+    Show (Tx era),
+    NoThunks (Tx era),
+    FromCBOR (Annotator (Tx era)),
+    ToCBOR (Tx era),
```

Comment:
Tim has proposed packaging these things up as `ChainData`, which would make this list somewhat cleaner.

Reply:
See the last commit.